presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetTypeUtils.java

```diff
@@ -14,17 +14,72 @@
 package com.facebook.presto.hive.parquet;
 
 import com.facebook.presto.hive.HiveColumnHandle;
+import parquet.column.ColumnDescriptor;
 import parquet.column.Encoding;
+import parquet.io.ColumnIO;
+import parquet.io.ColumnIOFactory;
+import parquet.io.InvalidRecordException;
 import parquet.io.ParquetDecodingException;
+import parquet.io.PrimitiveColumnIO;
 import parquet.schema.MessageType;
 import parquet.schema.Type;
 
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
 public final class ParquetTypeUtils
 {
     private ParquetTypeUtils()
     {
     }
 
+    public static List<PrimitiveColumnIO> getColumns(MessageType fileSchema, MessageType requestedSchema)
+    {
+        return (new ColumnIOFactory()).getColumnIO(requestedSchema, fileSchema, true).getLeaves();
+    }
+
+    public static RichColumnDescriptor getDescriptor(MessageType fileSchema, MessageType requestedSchema, List<String> path)
+    {
+        checkArgument(path.size() >= 1, "Parquet nested path should have at least one component");
+        int level = path.size();
+        for (PrimitiveColumnIO columnIO : getColumns(fileSchema, requestedSchema)) {
+            ColumnIO[] fields = columnIO.getPath();
+            if (fields.length <= level) {
+                continue;
+            }
+            if (fields[level].getName().equalsIgnoreCase(path.get(level - 1))) {
+                boolean match = true;
+                for (int i = 0; i < level - 1; i++) {
+                    if (!fields[i + 1].getName().equalsIgnoreCase(path.get(i))) {
+                        match = false;
+                    }
+                }
+
+                if (match) {
+                    ColumnDescriptor descriptor = columnIO.getColumnDescriptor();
+                    return new RichColumnDescriptor(descriptor.getPath(), columnIO.getType().asPrimitiveType(), descriptor.getMaxRepetitionLevel(), descriptor.getMaxDefinitionLevel());
+                }
+            }
+        }
+        return null;
```

Comment:
I suggest returning `Optional<RichColumnDescriptor>` here as I see not all the callers handle the case when this returns null. e.g., `ParquetPageSource` passes the return of this call to `ParquetBlockLoader`, which doesn't check for nulls and pass it to `parquetReader.readPrimitive`. Making this optional should catch all these places.